### ISSUE TITLE
Mousai says microphone is muted

### ISSUE THREAD

**Issue by apandada1** [SeaDve/Mousai#29]
OS: Manjaro XFCE
Installation method: AUR
Affected Version: 0.6.0 and 0.6.1

![image](https://user-images.githubusercontent.com/3063132/130635074-5108bb73-e4bd-4121-89a6-d5c62d0f935f.png)
Afterwards, it says "No audio detected" and "Please check your audio device". I have a working microphone in my laptop (it works with other apps, e.g. skype), and I also tried an earphone.

When I run from the command line, I get the error
```
$ mousai
Error: gst-resource-error-quark: Could not open file "/home/archisman/.cache/tmp/mousaitmp.ogg" for writing. (6) ../gstreamer/plugins/elements/gstfilesink.c(471): gst_file_sink_open_file (): /GstPipeline:pipeline0/GstFileSink:filesink0:
system error: Bad file descriptor
Error: gst-core-error-quark: GStreamer error: state change failed and some element failed to post a proper error message with the reason for the failure. (4) ../gstreamer/libs/gst/base/gstbasesink.c(5865): gst_base_sink_change_state (): /GstPipeline:pipeline0/GstFileSink:filesink0:
Failed to start
```

I cannot say when this issue appeared.

Last time I used the app was in July, and then the issue was absent.

**Comment by apandada1** [comment]
I installed previous versions one by one, and 0.4.4 is the last version that works in my computer (0.5.0 shows the same issue).

**Comment by SeaDve** [comment]
`/home/archisman/.cache/tmp` Does this directoey exist?

**Comment by SeaDve** [comment]
This is related to [this](https://github.com/SeaDve/Kooha/issues/61) I suppose

**Comment by SeaDve** [comment]
I made a release and it should also now come to aur

**Comment by SeaDve** [comment]
I shoud have implemented more helpful error messages 😅

**Comment by apandada1** [comment]
It is showing the same error message. I tried deleting `/home/archisman/.cache/tmp`, but that did not work.
It is creating an empty file `mousaitmp.ogg` in the `~/.cache/tmp` directory.
Also, now it does not show any error message in the terminal.
I also locally installed from source using meson and ninja, and the same error message exists.

I noticed that until 0.4.4 the AUR package used `/tmp` (not `.cache/tmp`). Is there a way to hardcode mousai to use `/tmp`?

**Comment by SeaDve** [comment]
> I noticed that until 0.4.4 the AUR package used `/tmp` (not `.cache/tmp`). Is there a way to hardcode mousai to always use `/tmp` instead on `.~/cache/tmp`?

You can modify get_tmp_dir() in utils directory

**Comment by SeaDve** [comment]
> I also locally installed from source using meson and ninja, and the same error message exists.

I think the issue is fixed then? There is btw a new option in the hamburger menu to change audio source if that change anything

**Comment by apandada1** [comment]
> I think the issue is fixed then? 
No, I meant, the same issue from AUR is present when I install from source.

> There is btw a new option in the hamburger menu to change audio source if that change anything

I will check that

**Comment by SeaDve** [comment]
> No, I meant, the same issue from AUR is present when I install from source.

I mean't the issue that the recorder doesn't detect the audio file, since it doesnt output any error anymore on the terminal.

(Oops i editted your reply instead of creating a new reply)

**Comment by apandada1** [comment]
Now it is working.
It seems that `Desktop Audio` was selected instead of microphone, and that is why it still said "muted".

I will do a few more tests, and close this issue in a few hours (till then, please keep it open).

**Comment by SeaDve** [comment]
Okay, thanks. You can close it yourself
